lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -96,93 +81,83 @@ describe('NxModal', function() {
-      const mockCallBack = jest.fn();
-      const component = getModal({ onCancel: mockCallBack });
+      await user.keyboard('[Tab]');
+      await user.keyboard('[Enter]');
+      await user.keyboard('q');
+      await user.keyboard('Q');
 
-      component.simulate('keyDown', createEvent('Tab'));
-      component.simulate('keyDown', createEvent('Enter'));
-      component.simulate('keyDown', createEvent('q'));
-      component.simulate('keyDown', createEvent('Q'));
       expect(mockCallBack).not.toHaveBeenCalled();
     });
 
-    it('calls stopPropagation and stopImmediatePropagation on Escape keydowns', function() {
-      const component = getModal({ onClose: jest.fn() }),
-          escEvent = createEvent(),
-          otherEvent = createEvent('q');
+    it('executes onCancel method ONLY when pressing ESC key', async function () {
+      const mockCallBack = jest.fn(),
+          user = userEvent.setup();
 
-      component.simulate('keyDown', escEvent);
-      component.simulate('keyDown', otherEvent);
+      quickRender({ onCancel: mockCallBack });
 
-      expect(escEvent.stopPropagation).toHaveBeenCalled();
-      expect(escEvent.nativeEvent.stopImmediatePropagation).toHaveBeenCalled();
+      await user.keyboard('[Tab]');
+      await user.keyboard('[Enter]');
+      await user.keyboard('q');
+      await user.keyboard('Q');
 
-      expect(otherEvent.stopPropagation).not.toHaveBeenCalled();
-      expect(otherEvent.nativeEvent.stopImmediatePropagation).not.toHaveBeenCalled();
+      expect(mockCallBack).not.toHaveBeenCalled();
     });
-  });
 
-  it('renders descendant tooltips attached to the backdrop rather than the document body', function() {
-    const nxModal = mount(
-      <NxModal onClose={() => {}}>
-        <div id="test-div">
-          <NxTooltip title="foo">
-            <NxButton>Foo</NxButton>
-          </NxTooltip>
+    it('does not bubble escape to outside listener', async function() {
+      const wrapperKeyDownListener = jest.fn(),
+          user = userEvent.setup();
+
+      render(
+        <div onKeyDown={wrapperKeyDownListener}>
+          <NxModal onClose={jest.fn()}>Hello</NxModal>
         </div>
-      </NxModal>
-    );
+      );
 
-    const tooltip = nxModal.find(Tooltip).at(0);
+      await user.keyboard('[Escape]');
+      expect(wrapperKeyDownListener).not.toHaveBeenCalled();
 
-    expect(tooltip.prop('PopperProps')!.container).toBe(nxModal.getDOMNode());
+      await user.keyboard('Q');
+      expect(wrapperKeyDownListener).toHaveBeenCalled();
+    });
   });
 
   it('moves focus back to the previously focused element when closed', async function() {
     function Fixture({ modalOpen }: { modalOpen: boolean }) {
       return (
         <>
-          <button id="test-btn">Test</button>
+          <button data-testid="test-btn">Test</button>
           { modalOpen && <NxModal onCancel={jest.fn()}><button id="cancel-btn">Close</button></NxModal> }
```

Comment:
I wonder if the `id="cancel-btn"` is needed and if not could be removed.

Reply:
> I wonder if the `id="cancel-btn"` is needed and if not could be removed.

Yup, can certainly be removed. df890b63f1dccfd8a2311c44c231a9bdbacd2ba2